pdf/pdfrenderthread.cpp

```diff
@@ -483,7 +483,12 @@ void PDFRenderThreadQueue::processPendingJob()
     case PDFJob::LoadDocumentJob:
         d->loadFailure = false;
         break;
+    case PDFJob::UnLockDocumentJob:
+        job->m_document = d->document;
+        break;
     default:
+        if (d->document->isLocked())
+            return;
```

Comment:
Ok, I've added a comment before the test. I've also added some qWarning() in pdfjob.cpp to signal something unusual happening (and avoid segfaults also).